new-app/src/App/runGame/InputGuess.tsx

```diff
@@ -1,37 +1,48 @@
 import React, {useState} from 'react';
-import * as Code from '../../model/number';
+import * as Code from '../../model/code';
 
 const InputGuess = () => {
-  const [guess, setGuess] = useState('');
+  const [guessString, setGuessString] = useState('');
+  const [stringForm, setStringForm] = useState('');
 
   const handleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
     const newGuess = event.target.value;
-    setGuess(newGuess);
+    setGuessString(newGuess);
   };
+  
+  const submitGuess = () : void => {
+    const codeForm: Code.Code = Code.fromString(guessString)
+    const stringForm: string = Code.codeToString(codeForm)
+    console.log({stringForm}, {codeForm})
+  }
 
-
-  const handleSubmit = (event: React.KeyboardEvent): void => {
+  const guessOnEnter = (event: React.KeyboardEvent): void => {
     if (event.key === 'Enter'){
-      console.log({guess})
-    try {
-      console.log({guess: Code.fromString(guess)})
-    } catch(e){
-      console.log(e)
-    }
+      submitGuess()
     }
   };
+  const guessOnButton = () : void => {
+    submitGuess()
+  }
+
+  const handleSubmit = () => {
+    guessOnButton()
+  }
    
   return(
     <div>
        <input
-        value={guess}
+        value={guessString}
         onChange={handleInput}
         placeholder="Enter a 4-digit number"
         className="input"
-        onKeyPress={handleSubmit}
+        onKeyPress={guessOnEnter}
```

Comment:
I might consider using a more explicit handler function pattern here. I think it will make this easier to understand. The idea is that you have certain events that happen as a result of user input and then you handle them explicitly. This should make your code easier to reason through.

```typescript
const handleOnKeyPress = (event) => {
  if (event.key === "Enter") {
    submitGuess()
  }
}

onKeyPress={handleOnKeyPress}
```